### PR TITLE
Fetch file using SCP

### PR DESCRIPTION
**Pull request by bedecarroll**
This PR modifies `scp_handler.py` to add functions to fetch files via SCP. Existing functions have been modified so that the direction semantics make sense. I've endeavored to make sure any existing scripts do not break by setting the default as the existing "put".

# Additions
* `scp_fetch_file` in `SCPConn`
* `fetch_file` and `verify_local_space` in `FileTransfer`

# Modifications
* In `FileTransfer` added `direction` attribute with a default of `'put'`
* Changed `__init__` in `FileTransfer` so that the `source` and `destination` make sense based on the direction
* Changed `remote_file_size` in `FileTransfer` so that it does not rely on the `dest_file` being the remote system
* Changed `remote_md5` in `FileTransfer` so that it does not rely on the `dest_file` variable

I've tried to mirror the functionality in both directions where appropriate.

### CODE REVIEW

**Comment by ktbyers** [comment]
@bedecarroll Can you rebase your fork? This will get it up to date so that only the changes you are making will show up in the commit.

The answer here shows you how to do this:

http://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository

Let me know if you need any assistance.

**Comment by bedecarroll** [comment]
Hi @ktbyers I hope I did it ok. I pulled in the latest `dev_1_0` branch, added my change back in and force pushed it back to my `fetch_file` branch. Have to admit the whole rebasing thing was a little bit hard as there were other commits and this seemed the easiest way. Hope I did the right thing!

Thank you!

File: netmiko/scp_handler.py [hunk 1]
```diff
@@ -106,6 +130,21 @@ def check_file_exists(self, remote_cmd=""):
         else:
             raise ValueError("Unexpected output from check_file_exists")
 
+    def remote_file_size(self, remote_cmd="", remote_file=None, delay_factor=8):
```
Comment (bedecarroll) [inline netmiko/scp_handler.py]: I set the `delay_factor` to eight here because this is the standard for the functions in `master`.

**Comment by ktbyers** [comment]
@bedecarroll Looks fine.

**Comment by ktbyers** [comment]
@bedecarroll Okay, I did a lot of work on this today (september 3 commits in the dev_1_0 branch). I still need to do more testing on it.

I integrated 'get' and 'put' direction into a bunch of methods (which I think contradicts what I had recommended earlier, but I thought it made more sense to do it this way when I dived into it).

Basically, you specify direction='get' or direction='put' on the creation of FileTransfer() object then the below methods will work based upon the direction:
verify_space_available()
check_file_exists()
compare_md5()
transfer_file()
verify_file()

Note, I haven't tested the get operations yet. Probably start testing these tomorrow or Monday.

**Pull request merged**